### DISCUSSION TITLE
How to increase http.request_rate to more than 10,000?

### DISCUSSION THREAD

**Discussion by sridharveeravalli** [artilleryio/artillery#2388] [Artillery]
Hello Yall,
I am trying to load test an API end point in aws that has a throttle limit of 10,000
In this regard, I am trying to run my scenario in yml file using fargate below options:
--count=40 --cpu=16 --memory=32

But, I am not event getting close to the intended request rate. I could see anywhere between 100-150 requests/second based on metrics output by artillery.

I am open to some suggestions on how to achieve.

Thanks much,
Sri

**Comment by hassy** [comment]
What does your `phases` config look like and what's the distribution of response times from the server? (`median`, `p95` and `p99`)? You may need to increase the number of VUs or the number of requests generated by each VU with a [`loop`](https://www.artillery.io/docs/reference/engines/http#loops).

The request rate (tracked by `http.request_rate`) depends greatly on the back-pressure from the server. This blog post provides more details: https://www.artillery.io/blog/load-testing-workload-models#:~:text=The%20only%20way,scenario%20like%20this.

**Reply by sridharveeravalli** [reply]
I have increased VU's to 800, cpu=16 and memory to 120 gb but the requests per second gets topped at about 120-150 requests per second. these are my phases: 
 - duration: 2
      arrivalRate: 20
      name: Warm up phase
    - duration: 4
      arrivalRate: 200
      name: ramp up phase

**Reply by sridharveeravalli** [reply]
Also, I have included loop in my flow to count 100 but still unable to see http.request_rate going up. Below are my metrics:
errors.ETIMEDOUT: .............................................................. 311
http.codes.200: ................................................................ 9024
http.downloaded_bytes: ......................................................... 3383625
http.request_rate: ............................................................. 107/sec
http.requests: ................................................................. 10083
http.response_time:
  min: ......................................................................... 1
  max: ......................................................................... 16878
  mean: ........................................................................ 463.8
  median: ...................................................................... 59.7
  p95: ......................................................................... 2465.6
  p99: ......................................................................... 6838
http.responses: ................................................................ 9024
http_request_pass: ............................................................. 8610
  min: ......................................................................... 1
  max: ......................................................................... 16878
  mean: ........................................................................ 481.8
  median: ...................................................................... 58.6
  p95: ......................................................................... 2566.3
  p99: ......................................................................... 7117
vusers.created: ................................................................ 2281
vusers.created_by_name.0: ...................................................... 2281
vusers.failed: ................................................................. 311

**Reply by sridharveeravalli** [reply]
Hi @hassy 
Knock Knock  :)
I did use the loop construct and was able to generate a higher request rate.
But, I do run into timeout errors with text "errors.ETIMEDOUT" though I have a 20 sec http timeout.
Any ideas how to avoid or what might be causing them?

**Reply by hassy** [reply]
that's great @sridharveeravalli. `ETIMEDOUT` means the server is not replying within the timeout window you set (20s in your case). That usually indicates that the service under test is overloaded.

**Reply by sridharveeravalli** [reply]
Follow-up questions:
1. Does the timeout occur for insufficient resources - cpu or memory? I did not see a easy way to find out how much was being consumed.
2. How to achieve higher burst rate with minimal number of nodes?
